docs/tutorials/craft-a-snap.rst

```diff
@@ -0,0 +1,549 @@
+need to define the snap's *apps* -- its programs that run as processes and services on
+the host.
+
+In the project file, the ``apps`` key decides all of the snap's apps. Pyfiglet has one
+main program -- the ``/bin/pyfiglet`` file we saw earlier.
+
+Add the following after the ``parts`` section:
+
+.. code-block:: yaml
+    :caption: snapcraft.yaml
+
+    apps:
+      ukuzama-pyfiglet:
+        command: bin/figlet
+
+As this is the *main* app -- in other words, the command we want to run when the user
+calls the snap by name -- it should match the snap name.
+
+The core of an app entry is its ``command`` key, which is the shell command that the
+snap calls on the host. It's a path to an executable inside the snap, and can contain
+arguments. It isn't strictly tied to any binary built by the snap. It could instead be,
+for example, a combination of POSIX-compatible commands.
+
+
+Test the snap
+-------------
+
+Let's repack the snap and try running it. First, run ``snapcraft pack`` again.
+
+Then, install the snap locally:
+
+.. code-block:: bash
+
+    snap install ukuzama-pyfiglet_0.1_amd64.snap --devmode --dangerous
+
+Normally, snapd prevents us from installing snaps that aren't vetted or confined. But,
+if we tell it we're comfortable with installing a snap with full system access and that
+doesn't declare itself as stable -- a potentially dangerous decision -- it will install.
+While we're crafting and especially debugging snaps, it's easiest to install them with
+these flags.
+
+At long last, let's try running our snap.
+
+.. code-block:: bash
+
+    ukuzama-pyfiglet hello, world!
+
+You should see the succesful result:
+
+.. terminal::
+
+    _          _ _                             _     _ _
+   | |__   ___| | | ___    __      _____  _ __| | __| | |
+   | '_ \ / _ \ | |/ _ \   \ \ /\ / / _ \| '__| |/ _` | |
+   | | | |  __/ | | (_) |   \ V  V / (_) | |  | | (_| |_|
+   |_| |_|\___|_|_|\___( )   \_/\_/ \___/|_|  |_|\__,_(_)
+                       |/
+
+FIGlet can draw with different typeface styles, too. It's a fun little command.
+
+.. terminal::
+    :input: ukuzama-pyfiglet -f smscript ciao, mondo!
+    :user: crafter
+    :host: home
+
+     _  o  _,   _              _         _|   _  |
+    /   | / |  / \_   /|/|/|  / \_/|/|  / |  / \_|
+    \__/|/\/|_/\_/o    | | |_/\_/  | |_/\/|_/\_/ o
+                  /
+
+
+Override the main part's build
+------------------------------
+
+If you inspect the source files in the pyfiglet source code, you'll notice that the font
+files are split into two directories, with the second directory containing all fonts
+with unaccounted licenses. The project has a separate Make recipe for combining these
+two directories. We committed to the project's Python build, so we can't access this
+second set of fonts. Since we're making this snap for personal testing purposes, let's
+see if we can preserve all the fonts in the snap.
```

Comment:
I'm slightly wary of this. Is this a bad idea?

Reply:
I'm fine with it!